plugin.js

```diff
@@ -66,10 +57,34 @@ function onReqHandlerWrapper (fastify, hook) {
       if (cookieHeader) {
         fastifyReq.cookies = fastify.parseCookie(cookieHeader)
       }
+      fastifyRes[kReplySetCookies] = new Map()
       done()
     }
 }
 
+function fastifyCookieOnSendHandler (fastifyReq, fastifyRes, payload, done) {
+  if (fastifyRes[kReplySetCookies].size) {
+    let setCookie = fastifyRes.getHeader('Set-Cookie')
+
+    if (!setCookie) {
+      setCookie = []
+    }
+
+    if (typeof setCookie === 'string') {
+      setCookie = [setCookie]
+    }
+
+    for (const [, c] of fastifyRes[kReplySetCookies]) {
```

Comment:
Nitpick:
@mcollina 
Is this for of const of  of the Map ok?
Is there a perf difference to for const of Map.values()?

Reply:
I think it's better to use `.values()`, it'd likely be more performant.

Reply:
Something like this?

```js
function fastifyCookieOnSendHandler (fastifyReq, fastifyRes, payload, done) {
  if (fastifyRes[kReplySetCookies].size) {
    let setCookie = fastifyRes.getHeader('Set-Cookie')

    if (!setCookie) {
      setCookie = []
    }

    if (typeof setCookie === 'string') {
      setCookie = [setCookie]
    }

    const setCookiesIter = fastifyRes[kReplySetCookies].values()

    for (let i = 0; i < fastifyRes[kReplySetCookies].size; ++i) {
      const c = setCookiesIter.next().value

      setCookie.push(cookie.serialize(c.name, c.value, c.opts))
    }

    fastifyRes.removeHeader('Set-Cookie')
    fastifyRes.header('Set-Cookie', setCookie)
  }

  done()
}
```

Reply:
Like this?

```js
function fastifyCookieOnSendHandler (fastifyReq, fastifyRes, payload, done) {
  if (fastifyRes[kReplySetCookies].size) {
    let setCookie = fastifyRes.getHeader('Set-Cookie')

    if (!setCookie) {
      setCookie = []
    }

    if (typeof setCookie === 'string') {
      setCookie = [setCookie]
    }

    const setCookiesIter = fastifyRes[kReplySetCookies].values()

    for (let i = 0; i < fastifyRes[kReplySetCookies].size; ++i) {
      const c = setCookiesIter.next().value

      setCookie.push(cookie.serialize(c.name, c.value, c.opts))
    }

    fastifyRes.removeHeader('Set-Cookie')
    fastifyRes.header('Set-Cookie', setCookie)
  }

  done()
}
```

Reply:
The thing is, that `for const of` handles iterators. E.g. Object.keys() actually returns an iterator. But by doing const blaKeys = Object.keys(bla) node is resolving the iterator to an Array. If you have a huuuuge Array, you maybe dont want to store them temporary in a variable but just iterating them. 

for const of the Map is basically iterating over .entries() of the Map. by specifying .values() you dont get an array with key and value, but only the value. So you dont have to destructure the array and throw away the key. 

You dont need to do the next call manually. Just do

```suggestion
    for (const c of fastifyRes[kReplySetCookies].values()) {
```

Not tested.

Reply:
Maybe do the examples files below and test the perf and then check if values is faster.